### ISSUE TITLE
ci tests supervisor start failure

### ISSUE THREAD

**Issue by e-carlin** [radiasoft/sirepo#5627]
https://github.com/radiasoft/sirepo/actions/runs/4567494832/jobs/8061357768

**Comment by robnagler** [comment]
No they shouldn't be the same.

**Comment by robnagler** [comment]
Been getting more failures:

```txt
tests/partial_animation_test.py::test_synergia Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20210211_add_flash_proprietary_lib_files
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20210211_upgrade_runner_to_job_db
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20210218_add_flash_proprietary_lib_files_force
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20210301_migrate_role_jupyterhub
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20220609_add_expiration_column_to_user_role_t
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20220901_migrate_ml_to_activait
Apr 03 20:19:54 16187     0 ../../sirepo/db_upgrade.py:40:do_all running upgrade _20230203_drop_spa_session
 * Serving Flask app 'sirepo.server' (lazy loading)
 * Environment: development
 * Debug mode: off
 * Running on http://127.0.0.1:10538/ (Press CTRL+C to quit)
 * Restarting with stat
Apr 03 20:19:54 16190     0 ../../sirepo/job_driver/__init__.py:402:init_module modules=['local']
Apr 03 20:19:54 16190     0 ../../sirepo/pkcli/job_supervisor.py:78:default_command ip=127.0.0.1 port=10349
ERROR

==================================== ERRORS ====================================
_______________________ ERROR at setup of test_synergia ________________________
Traceback (most recent call last):
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 39, in fc_module
    with _subprocess_start(request, fc_args=a) as c:
  File "/home/vagrant/.pyenv/versions/3.7.2/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 395, in _subprocess_start
    _ping_supervisor(c.http_prefix + "/job-supervisor-ping")
  File "/home/runner/work/sirepo/sirepo/tests/conftest.py", line 378, in _ping_supervisor
    pkunit.pkfail("start failed uri={} exception={}", uri, l)
  File "/home/vagrant/.pyenv/versions/3.7.2/envs/py3/lib/python3.7/site-packages/pykern/pkunit.py", line 394, in pkfail
    raise PKFail("{} {}".format(call, msg))
pykern.pkunit.PKFail: tests/conftest.py:378:_ping_supervisor start failed uri=http://127.0.0.1:10538/job-supervisor-ping exception=HTTPConnectionPool(host='127.0.0.1', port=10538): Max retries exceeded with url: /job-supervisor-ping (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f6aa9d7c908>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

Note that the connection is failing for flask, not the supervisor. There should be an output from flask before it sends the message:

```txt
Apr 03 19:40:24 27282     2 ../../../sirepo/job_api.py:257:_content_and_uri api=api_jobSupervisorPing runDir=None
```

The port is being opened, and flask is not complaining it can't bind to port. 

One approach would be to have `pykern.pkcli.test` look at the log, and restart if running in a github action and the test is failing for this specific reason.  An alternative is to have conftest exit with KeyboardInterrupt which pkcli.test would see as [`exit(2)`](https://docs.pytest.org/en/6.2.x/usage.html), and then check if it was a github action. I like this better, because the test is loosely coupled and could be used in other cases.

**Comment by robnagler** [comment]
Leaving open because may add this: https://github.com/radiasoft/pykern/issues/350

**Comment by robnagler** [comment]
I think this is fixed, but maybe #5643 will be needed.
